script.js

```diff
@@ -0,0 +1,200 @@
+let webBrands = [];
+let brands=[];
+let resultbrands=[];
+let productNode;
+const proxyurl = "https://cors-anywhere.herokuapp.com/";
+const url = `https://wellspent-ethical.herokuapp.com/api/brands`;
+
+
+var linksList = document.querySelectorAll("[id^=product-]");
+linksList.forEach(function(product) {
+  const node = product.childNodes[0];
+  const label = node.getAttribute("aria-label");
+  let brand = label.toLowerCase();
+  webBrands.push(brand);
+});
+
+
+ // site that doesn’t send Access-Control-*
+fetch(proxyurl + url) // https://cors-anywhere.herokuapp.com/https://example.com
+  .then(resultbrandsonse => resultbrandsonse.text())
+  .then(contents => {
+    resultbrands = JSON.parse(contents);
+    brands = resultbrands;
+
+
+    const finalResult = brands.result.map((data)=>{
+    const result =   webBrands.filter((webBrand)=>{
+      const splitedBrand= data.BrandName.toLowerCase().split(" ")[0];
+      if(webBrand.indexOf(splitedBrand)==0){
+        return webBrand.includes(data.BrandName.toLowerCase())
+      }
+      })
+      return result.length > 0 ? [data.BrandName] : [];
+    });
+    const filteredBrand = Promise.all(finalResult)
+    .then((res)=>{
+        res.map((brand,index, array) => {
+          if(brand.length > 0){
+          linksList.forEach(function(product) {
+            const node = product.childNodes[0];
```

Comment:
Would be better if you put the code related to creating an HTML element using JS in a different function. So it will return the element. That will make a familiar code to be understood by anyone.